### PR TITLE
Improved error reporting when reading multiple JSON files

### PR DESCRIPTION
**Pull request by vuule**
## Description
`read_json` implementation cannot read multiple inputs when inputs are compressed or not in JSON lines format.
This PR adds explicit checks to return a clear error messages in unsupported cases.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/src/io/json/reader_impl.cu [hunk 1]
```diff
@@ -601,6 +601,8 @@ table_with_metadata read_json(std::vector<std::unique_ptr<datasource>>& sources,
   }
 
   CUDF_EXPECTS(not sources.empty(), "No sources were defined");
+  CUDF_EXPECTS(sources.size() == 1 or reader_opts.get_compression() == compression_type::NONE,
+               "Multiple compressed inputs are not supported");
```
Comment (ttnghia) [inline cpp/src/io/json/reader_impl.cu:605]: Nit: Please remove the empty line below this.

**Review by ttnghia — APPROVED** [review]

File: cpp/src/io/json/experimental/read_json.cpp [hunk 2]
```diff
@@ -161,6 +161,13 @@ table_with_metadata read_json(host_span<std::unique_ptr<datasource>> sources,
                  "specifying a byte range is supported only for json lines");
```
Comment (bdice) [inline cpp/src/io/json/experimental/read_json.cpp:161]: Can we standardize this error message with the other one added in this PR? Generally, we should be capitalizing JSON and decide whether to use `lines` or `Lines` consistently across the codebase.
```suggestion
                 "specifying a byte range is supported only for JSON Lines");
```

**Review by bdice — APPROVED** [review]
One comment about error message / naming consistency.

**Review by hyperbolic2346 — APPROVED** [review]

**Comment by vuule** [comment]
@gpucibot merge

**Pull request merged**